plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/OpenShiftDebugUtils.java

```diff
@@ -155,36 +171,99 @@ public void done(IJobChangeEvent event) {
 
 	private void updateDeploymentConfigValues(IDeploymentConfig deploymentConfig,
 			DebuggingContext debugContext) {
-		Collection<IContainer> originalContainers = deploymentConfig.getContainers();
-		if (originalContainers != null && !originalContainers.isEmpty() && debugContext.isDebugEnabled()) {
-			Collection<IContainer> containers = new ArrayList<>(originalContainers);
-			IContainer container = containers.iterator().next();
-			Set<IPort> ports = new HashSet<>(container.getPorts());
-			
-			IPort existing = ports.stream().filter(p -> p.getContainerPort() == debugContext.getDebugPort())
-										.findFirst().orElse(null);
-			boolean added = false;
-			if (existing == null) {
-				PortSpecAdapter newPort = new PortSpecAdapter("debug", "TCP", debugContext.getDebugPort());
-				added = ports.add(newPort);
-			} else {
-				PortSpecAdapter newPort = new PortSpecAdapter(existing.getName(), existing.getProtocol(), debugContext.getDebugPort());
-				if (!existing.equals(newPort)) {
-					ports.remove(existing);
+		if(debugContext.isDebugEnabled()) {
```

Comment:
actually, because debugContexts.setEnabled(false) was set L123, the logic must be reverted here, else we go into the other branch, set the new deployment config to debug mode and then wait for pod redeployment until timeout (no pod is redeployed because deployment config hasn't really changed)